integrationtests/agent/watcher_trigger_test.go

```diff
@@ -0,0 +1,194 @@
+package agent
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/utils/pointer"
+
+	"github.com/rancher/fleet/internal/cmd/agent/trigger"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("Watches for deployed resources", Ordered, func() {
+
+	var (
+		env    *specEnv
+		triggr *trigger.Trigger
+	)
+
+	BeforeAll(func() {
+		env = specEnvs["watchertrigger"]
+		triggr = trigger.New(ctx, env.k8sClient.RESTMapper(), dynamic.NewForConfigOrDie(cfg))
+		DeferCleanup(func() {
+			Expect(k8sClient.Delete(ctx, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: env.namespace}})).ToNot(HaveOccurred())
+		})
+	})
+
+	registerResource := func(key string, objs ...runtime.Object) *int {
+		_ = triggr.Clear(key)
+		var count int
+		Expect(triggr.OnChange(key, env.namespace, func() {
+			count++
+		}, objs...)).ToNot(HaveOccurred())
+		return &count
+	}
+
+	When("watching a deployed configmap", Ordered, func() {
+		createConfigMap := func() *corev1.ConfigMap {
+			cm := corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "test-cm-",
+					Namespace:    env.namespace,
+				},
+			}
+
+			err := env.k8sClient.Create(ctx, &cm)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(cm.UID).ToNot(BeEmpty())
+			return &cm
+		}
+
+		var triggerCount *int
+		var cm *corev1.ConfigMap
+		BeforeEach(func() {
+			cm = createConfigMap()
+			triggerCount = registerResource(env.namespace+"/test-configmap", cm)
+			DeferCleanup(func() {
+				Expect(
+					client.IgnoreNotFound(env.k8sClient.Delete(ctx, cm))).
+					NotTo(HaveOccurred())
+			})
+		})
+		It("is not initially triggered", func() {
+			Consistently(func() int {
+				return *triggerCount
+			}).Should(Equal(0))
+		})
+		It("is triggered on deletion", func() {
+			Expect(env.k8sClient.Delete(ctx, cm)).ToNot(HaveOccurred())
+			Eventually(func() int {
+				return *triggerCount
+			}).WithPolling(100 * time.Millisecond).MustPassRepeatedly(3).
```

Comment:
Out of curiosity, where does this `3` come from?

Reply:
It's a way to express that the trigger count should eventually increase from 0 -> 1, and not increase any longer. In other words, just executed once.
The actual 3 and the polling interval are totally made up values, I didn't want to slow down the tests too much either.
Happy to apply any suggestions you may have!

Reply:
No worries, I was just wondering how you came to `3` instead of, say, 5 or 10 (for instance, also made up values).
3 works for me :)